sim/src/recorder.rs

```diff
@@ -30,33 +30,31 @@ impl TrafficRecorder {
     }
 
     pub fn handle_event(&mut self, time: Time, ev: &Event, map: &Map, driving: &DrivingSimState) {
-        if let Event::AgentEntersTraversable(a, Some(trip), on, _) = ev {
```

Comment:
Wow, this really was an uncomfortable amount of nesting...

Reply:
swift has a nice "guard statement" syntax that addresses this kind of nesting... the best I know of in rust is something like:

```
let (car, trip, on) = match ev {
    Event::AgentEntersTraversable(AgentID::Car(car), Some(trip), on, _) => (car, trip, on),
    _ => return
}
if self.seen_trips.contains(&trip) {
    return;
}
let l = match on {
  Traversable::Lane(l) => l,
   _ => return;
}
```

Reply:
Yeah, it might be possible to rewrite it as match statements with if guards, or to use chained if lets (which requires nightly).